controllers/datadogagent/component/agent/default.go

```diff
@@ -307,6 +307,25 @@ func envVarsForCoreAgent(dda metav1.Object) []corev1.EnvVar {
 	return append(envs, commonEnvVars(dda)...)
 }
 
+func envVarsForTraceAgent(dda metav1.Object) []corev1.EnvVar {
```

Comment:
Is this the right place to add new env variables for Trace Agent? Looks like this codepath is not covered by agent_test.go tests.